src/main/java/school/hei/haapi/repository/PaymentRepository.java

```diff
@@ -13,7 +13,8 @@ public interface PaymentRepository extends JpaRepository<Payment, String> {
   @Query(
       value =
           "select p from Payment p join Fee f on f.id = p.fee.id"
-              + " where f.student.id = :student_id and f.id = :fee_id")
+              + " where f.student.id = :student_id and f.id = :fee_id"
+              + " and p.isDeleted = false")
```

Comment:
test without adding isDeleted check in the sql request.

Normally the @Where decorator already checks this field's value